client/my-sites/site-monitoring/metrics-tab.tsx

```diff
@@ -66,20 +72,77 @@ export function useSiteMetricsData( timeRange: TimeRange, metric?: MetricsType )
 
 	// Process the data in the format accepted by uPlot
 	const formattedData =
-		data?.data?.periods?.reduce(
-			( acc, period ) => {
-				acc[ 0 ].push( period.timestamp );
-				acc[ 1 ].push( getDimensionValue( period ) );
+		requestsData?.data?.periods?.reduce(
+			( acc, period, index ) => {
+				const timestamp = period.timestamp;
+
+				// Check if the timestamp is already in the arrays, if not, push it
+				if ( acc[ 0 ][ acc[ 0 ].length - 1 ] !== timestamp ) {
+					acc[ 0 ].push( timestamp );
+					acc[ 1 ].push( getDimensionValue( period ) ); // Blue line data
+
+					// Add response time data as a green line
+					if ( responseTimeData?.data?.periods && responseTimeData.data.periods[ index ] ) {
+						acc[ 2 ].push( getDimensionValue( responseTimeData.data.periods[ index ] ) );
+					}
+				}
+
 				return acc;
 			},
-			[ [], [] ] as Array< Array< number | null > > // Define the correct initial value type
-		) || ( [ [], [] ] as Array< Array< number | null > > ); // Return a default value when data is not available yet
+			[ [], [], [] ] as Array< Array< number | null > > // Adjust the initial value with placeholders for both lines
+		) || ( [ [], [], [] ] as Array< Array< number | null > > ); // Return default value when data is not available yet
 
 	return {
 		formattedData,
 	};
 }
 
+export function useSiteMetricsData400vs500(
+	timeRange: TimeRange,
+	metric?: MetricsType,
+	dimension?: DimensionParams
+) {
+	const siteId = useSelector( getSelectedSiteId );
+
+	// Use the custom hook for time range selection
+	const { start, end } = timeRange;
+
+	const { data } = useSiteMetricsQuery( siteId, {
+		start,
+		end,
+		metric: metric || 'requests_persec',
+		dimension: dimension || 'http_status',
+	} );
+
+	const formattedDataHTTP = data?.data?.periods?.reduce(
+		( acc, period ) => {
+			const timestamp = period.timestamp;
+
+			// Check if the timestamp is already in the arrays, if not, push it
+			if ( acc[ 0 ][ acc[ 0 ].length - 1 ] !== timestamp ) {
+				acc[ 0 ].push( timestamp );
+
+				// Check if the dimension object contains values for 400 and 500 status codes
+				if ( period.dimension && ( period.dimension[ '400' ] || period.dimension[ '500' ] ) ) {
```

Comment:
We need to check all `4xx` and `5xx` responses, not just `400` and `500`.

Reply:
Not needed anymore since we changed the direction supporting multilines.